woocommerce/woocommerce-ios
# Blaze: Hide Blaze button on the product form after campaign creation

Closes: #11562 


## Description

This PR updates the product form to hide the Blaze button after campaign creation succeeds. Changes include:
- `ProductFormViewController`: Resync Blaze campaign list after creating a Blaze campaign to update storage items and the table view.
- `ProductFormViewModel`: 
  - Updated the result controller to fetch campaigns by campaign ID.
  - Update action factory when there are any updates to storage items.
  - Removed the redundant `blazeEligibilityUpdate` publisher since blaze eligibility check has been updated to be synchronous.

## Testing steps

- Log in to a store eligible for Blaze.
- Navigate to the Products tab and select an existing a product or publish a new one.
- Tap on Promote with Blaze and follow the steps to create a campaign for the product.
- After campaign creation completes, confirm that the Promote with Blaze button is hidden.

## Screenshots


https://github.com/user-attachments/assets/1bbcd649-58f7-45d2-8cae-45f285fea261



---
